packages/react-shopify-app-bridge/src/Provider.tsx

```diff
@@ -152,7 +163,14 @@ const InnerGadgetProvider = memo(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [gadgetAppUrl, isRootFrameRequest, originalQueryParams, redirectToOauth]);
 
-    const loading = forceRedirect || redirectToOauth || fetchingOrInstallingShop || !hasStartedFetchingOrInstallingShop.current;
+    // we're loading if:
+    //  - we need to redirect somewhere (forceRedirect || redirectToOauth)
+    //  - if we don't have an install state hint, and we're loading the shop from the backend to check current state
+    //  - if we do have an install state hint, and the hint tells us we need to re-authenticate
+    const loading =
+      forceRedirect ||
+      redirectToOauth ||
+      (hasInstallStateHint ? !isAuthenticated : fetchingOrInstallingShop || !hasStartedFetchingOrInstallingShop.current);
```

Comment:
if there is a hint and the mutation comes back with `isAuthenticated = false` I think this means that we will still be in a loading state?

Reply:
Good call, I will update this to respect the value from the mutation regardless of what the hint says.